internal/service/sso/data_source_application.go

```diff
@@ -0,0 +1,1166 @@
+							Description: "A string that specifies the logout endpoint URL.",
+							Computed:    true,
+						},
+						"slo_response_endpoint": schema.StringAttribute{
+							Description: "A string that specifies the endpoint URL to submit the logout response.",
+							Computed:    true,
+						},
+						"slo_window": schema.Int64Attribute{
+							Description: "An integer that defines how long (hours) PingOne can exchange logout messages with the application, specifically a logout request from the application, since the initial request.",
+							Computed:    true,
+						},
+						"sp_entity_id": schema.StringAttribute{
+							Description: "A string that specifies the service provider entity ID used to lookup the application. This is a required property and is unique within the environment.",
+							Computed:    true,
+						},
+						"sp_verification_certificate_ids": schema.SetAttribute{
+							Description: "A list that specifies the certificate IDs used to verify the service provider signature.",
+							ElementType: types.StringType,
+							Computed:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *ApplicationDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	resourceConfig, ok := req.ProviderData.(framework.ResourceType)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected the provider client, got: %T. Please report this issue to the provider maintainers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.Client = resourceConfig.Client.API
+	if r.Client == nil {
+		resp.Diagnostics.AddError(
+			"Client not initialised",
+			"Expected the PingOne client, got nil.  Please report this issue to the provider maintainers.",
+		)
+		return
+	}
+}
+
+func (r *ApplicationDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data *applicationDataSourceModel
+
+	if r.Client.ManagementAPIClient == nil {
+		resp.Diagnostics.AddError(
+			"Client not initialized",
+			"Expected the PingOne client, got nil.  Please report this issue to the provider maintainers.")
+		return
+	}
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var application management.ReadOneApplication200Response
+
+	// Application API does not support SCIM filtering
+	if !data.ApplicationId.IsNull() {
+		// Run the API call
+		var response *management.ReadOneApplication200Response
+		resp.Diagnostics.Append(framework.ParseResponse(
+			ctx,
+
+			func() (any, *http.Response, error) {
+				return r.Client.ManagementAPIClient.ApplicationsApi.ReadOneApplication(ctx, data.EnvironmentId.ValueString(), data.ApplicationId.ValueString()).Execute()
```

Comment:
This needs to have an "environment exists" check added because the API doesn't respond with the correct http code if the environment has been removed.  Example:
https://github.com/pingidentity/terraform-provider-pingone/blob/677191a70df4229ca33f1db91d2503e09746800b/internal/service/sso/resource_application.go#L728-L729